### PR TITLE
Project Whitelist

### PR DESCRIPTION
**Pull request by nicksalt**
- Adds project whitelist model
- Whitelist can be configured for each project from the Admin page
- Commits squashed into one after some migrations issues
- Full test coverage for new Project Whitelist service

### CODE REVIEW

File: Portal/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs [hunk 1]
```diff
@@ -192,6 +193,13 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                   .WithOne(e => e.MonthlyUsage)
                   .OnDelete(DeleteBehavior.NoAction);
         });
+        modelBuilder.Entity<Project_Whitelist>(entity =>
+        {
+            entity.HasKey(e => e.Id);
+            entity.HasOne(e => e.Project)
+                .WithOne(e => e.Whitelist)
+                .OnDelete(DeleteBehavior.NoAction);
```
Comment (ErikApption) [inline Portal/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs:201]: Any reason we have this OnDelete behavior

File: Portal/src/Datahub.Portal/i18n/localization.fr.json [hunk 2]
```diff
@@ -1688,13 +1688,16 @@
...
+      "WhitelistProjectResources": "Ressources du projet de liste blanche"
+
     },
     "CARD_DESCRIPTIONS": {
       "ExternalSharing": "Gérer le partage externe des fichiers et des rapports Power BI",
       "ManageServices": "Approuver la création de services et les demandes d'accès",
       "Registrations": "Gérer les enregistrements",
       "SharingAPIs": "Gérer les clés API pour les utilisateurs externes",
       "Metadata": "Gérer le modèle de métadonnées de Datahub",
+      "WhitelistProjectResources": "Ressources du projet de liste blanche"
```
Comment (ErikApption) [inline Portal/src/Datahub.Portal/i18n/localization.fr.json:1700]: Gérer les ressources auxquelles l'espace de travail a accès

**Review by ErikApption — CHANGES_REQUESTED** [review]
Please fix the french translation

File: Portal/src/Datahub.Portal/i18n/localization.fr.json (hunk 2 above)
Comment (nicksalt) [reply Portal/src/Datahub.Portal/i18n/localization.fr.json:1700]: thanks, fixed in commit 374ce84024c2adfa607377231094e98b9ee48b1b

File: Portal/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs (hunk 1 above)
Comment (nicksalt) [reply Portal/src/Datahub.Core/Model/Datahub/DatahubProjectDBContext.cs:201]: No particular reason here, I was working with @iroelperez and running into some weird migration issues. We were trying to make sure there were no differences between the project whitelist and projected credits so we could rule out anything... I think this is the default behaviour though

**Pull request merged**